Sources/Basics/GitURL.swift

```diff
@@ -0,0 +1,44 @@
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+public struct GitURL: Encodable, Equatable, Hashable, Sendable {
```

Comment:
should this be SourceControlURL (sorry)

Reply:
Hm, not sure. On the one hand, we are doing this specifically for git, so seems right to make it explicit. On the other hand, this does show up in a lot of APIs that are theoretically not git specific forever.

Reply:
We decided to go with `SourceControlURL` here.